README.md

```diff
@@ -894,239 +912,207 @@ How to test actions with scenario:
-
-```python
-
-from scenario import Relation
-
-
-class MyCharm(...):
-    ...
+    # So many other changes have been to avoid these string names, so this seems awkward.
+    assert state_out.deferred[0].name == 'start'
 
+    # Could we make this work?
+    assert isinstance(state_out.deferred[0], ops.StartEvent)
 
-foo_relation = Relation('foo')
-foo_relation.changed_event.deferred(handler=MyCharm._on_foo_relation_changed)
+    # Or just this?
+    assert state_out.deferred[0].event is ops.StartEvent
 ```
 
 ### Fine-tuning
 
-The deferred helper Scenario provides will not support out of the box all custom event subclasses, or events emitted by
+The `deferred` helper Scenario provides will not support out of the box all custom event subclasses, or events emitted by
 charm libraries or objects other than the main charm class.
 
 For general-purpose usage, you will need to instantiate DeferredEvent directly.
 
 ```python
-from scenario import DeferredEvent
+import scenario
 
-my_deferred_event = DeferredEvent(
+my_deferred_event = scenario.DeferredEvent(
     handle_path='MyCharm/MyCharmLib/on/database_ready[1]',
     owner='MyCharmLib',  # the object observing the event. Could also be MyCharm.
     observer='_on_database_ready'
 )
 ```
 
-# StoredState
+## StoredState
+
+TONY NOTE: this belongs up in the state section
 
 Scenario can simulate StoredState. You can define it on the input side as:
 
 ```python
-from ops.charm import CharmBase
-from ops.framework import StoredState as Ops_StoredState, Framework
-from scenario import State, StoredState
+import ops
+import scenario
 
 
-class MyCharmType(CharmBase):
-    my_stored_state = Ops_StoredState()
+class MyCharm(ops.CharmBase):
+    my_stored_state = ops.StoredState()
 
-    def __init__(self, framework: Framework):
+    def __init__(self, framework):
         super().__init__(framework)
         assert self.my_stored_state.foo == 'bar'  # this will pass!
 
 
-state = State(stored_state=[
-    StoredState(
-        owner_path="MyCharmType",
-        name="my_stored_state",
+# This feels much like containers - it doesn't really make sense for it to be a list.
+state = scenario.State(stored_state=[
+    scenario.StoredState(
+        charm="MyCharm",
+        attr="my_stored_state",
         content={
             'foo': 'bar',
             'baz': {42: 42},
         })
 ])
+
+# Similar to containers, a set won't work now, but I think these are meant to be
+# immutable (even the content) so in theory could, and are definitely simpler than containers.
+state = scenario.State(stored_state={
+    scenario.StoredState(
+        charm="MyCharm",
+        attr="my_stored_state",
+        content={
+            'foo': 'bar',
+            'baz': {42: 42},
+        })
+})
+
+# Move the charm and attribute to a key? This also avoids having two objects in the list that
+# have the same destination.
+state = scenario.State(stored_state={
+    "MyCharm.mystored_state": scenario.StoredState({
```

Comment:
Feels weird to have a dotted name here. Why is the charm necessary -- isn't there only ever one charm?

Additionally, does `StoredState` need an `.attr` attribute, and if so, can `scenario.State()` populate it automatically?